vendor/kops-1.12/Makefile

```diff
@@ -3,6 +3,8 @@ export MAJOR_VERSION = 1.12
 export PACKAGE_NAME = $(MASTER_PACKAGE_NAME)-$(MAJOR_VERSION)
 export PACKAGE_REPO_NAME = kops
 export DOWNLOAD_URL = https://github.com/kubernetes/kops/releases/download/$(PACKAGE_VERSION)/kops-$(OS)-$(ARCH)
+export PACKAGE_VERSION_PIN = $(MAJOR_VERSION).x
```

Comment:
Since 99% of the time we pin to the `$(MAJOR_VERSION).x`, my preference would be:

`tasks/Makefile.package`:
```
ifneq ($(MAJOR_VERSION),,)
export PACKAGE_VERSION_PIN ?= $(MAJOR_VERSION).x
endif
```
Which conditionally sets `PACKAGE_VERSION_PIN` (if not already set), and only when `MAJOR_VERSION` is not empty.

This way we don't need to always set `MAJOR_VERSION` and `PACKAGE_VERSION_PIN`, but still have the ability to overwrite `PACKAGE_VERSION_PIN`

Reply:
I don't like this for a few reasons.
1. Even though I used a conditional in this PR, to select `PACKAGE_VERSION_TARGET`, I do not generally like conditionals in Makefiles because they alter the Makefile based on the values set when the Makefile is read, unlike regular variables which can be evaluated after everything is read in and based on the target that is executing. I would not have used a conditional at all except that I wanted to be extra careful not to break the existing normal auto-update by modifying it too much.
2. We use `MAJOR_VERSION=latest` for unpinned packages when we also have pinned packages, and there would not be an opportunity to unset `PACKAGE_VERSION_PIN` between the time it is set by this new conditional and the time it is evaluated by the target selection conditional.
3. I feel it is not worth the extra complication and fragility to save 1 line of boilerplate code, especially given how much boilerplate is already copy/pasted for the post-install and post-deinstall scripts.